src/lints/manual/manual_assert.rs

```diff
@@ -0,0 +1,123 @@
+use cairo_lang_defs::{ids::ModuleItemId, plugin::PluginDiagnostic};
+use cairo_lang_diagnostics::Severity;
+use cairo_lang_semantic::{db::SemanticGroup, Arenas, Expr, ExprBlock, ExprIf, Statement};
+use cairo_lang_syntax::node::TypedStablePtr;
+use if_chain::if_chain;
+
+use crate::{
+    context::{CairoLintKind, Lint},
+    helper::is_panic_expr,
+    queries::{get_all_function_bodies, get_all_if_expressions},
+};
+
+pub struct ManualAssert;
+
+/// ## What it does
+///
+/// Checks for manual implementations of `assert` macro in `if` expressions.
+///
+/// ## Example
+///
+/// ```cairo
+/// fn main() {
+///     let a = 5;
+///     if a == 5 {
+///         panic!("a shouldn't be equal to 5");
+///     }
+/// }
+/// ```
+///
+/// Can be rewritten as:
+///
+/// ```cairo
+/// fn main() {
+///     let a = 5;
+///     assert!(a != 5, "a shouldn't be equal to 5");
+/// }
+/// ```
+impl Lint for ManualAssert {
+    fn allowed_name(&self) -> &'static str {
+        "manual_assert"
+    }
+
+    fn diagnostic_message(&self) -> &'static str {
+        "Manual assert detected. Consider using assert!() macro instead."
+    }
+
+    fn kind(&self) -> crate::context::CairoLintKind {
+        CairoLintKind::ManualAssert
+    }
+}
+
+pub fn check_manual_assert(
+    db: &dyn SemanticGroup,
+    item: &ModuleItemId,
+    diagnostics: &mut Vec<PluginDiagnostic>,
+) {
+    let function_bodies = get_all_function_bodies(db, item);
+    for function_body in function_bodies.iter() {
+        let if_exprs = get_all_if_expressions(function_body);
+        let arenas = &function_body.arenas;
+        for if_expr in if_exprs.iter() {
+            check_single_manual_assert(db, if_expr, arenas, diagnostics);
+        }
+    }
+}
+
+fn check_single_manual_assert(
+    db: &dyn SemanticGroup,
+    if_expr: &ExprIf,
+    arenas: &Arenas,
+    diagnostics: &mut Vec<PluginDiagnostic>,
+) {
+    let Expr::Block(ref if_block) = arenas.exprs[if_expr.if_block] else {
+        return;
+    };
+
+    check_single_condition_block(db, if_block, if_expr, arenas, diagnostics);
+
+    if_chain! {
+      if let Some(else_block) = if_expr.else_block;
+      if let Expr::Block(ref else_block) = arenas.exprs[else_block];
+      then {
+        check_single_condition_block(db, else_block, if_expr, arenas, diagnostics);
+      }
```

Comment:
Can we add tests for those? 🙏

Reply:
https://github.com/software-mansion/cairo-lint/blob/a9af137d86ff6d31e40c0d942db962266d14c992/tests/manual/manual_assert.rs#L136 there is already

Reply:
mea culpa